server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java

```diff
@@ -92,20 +94,23 @@ public E get(int index) {
 
     @Override
     public Iterator<E> iterator() {
-        return new SafeIterator<E>(super.iterator());
+        return new SafeIterator<>(super.iterator());
     }
 
     @Override
     public List<E> subList(int fromIndex, int toIndex) {
         if (fromIndex < 0 || toIndex > size || fromIndex > toIndex) {
             throw new IndexOutOfBoundsException("The indexes provided are outside the bounds of this list.");
         }
-        return new SafeIteratorList<E>(super.subList(fromIndex, toIndex));
+        return new SafeIteratorList<>(super.subList(fromIndex, toIndex));
     }
 
     private String getCountSql(String sql) {
-        String result = sql.replaceAll("(?i)select (.*?) from (.*)", "select count(*) from $2");
-        int orderByPos = result.toLowerCase().lastIndexOf("order by");
+        String result = UaaStringUtils.getValidatedString(sql).toLowerCase();
```

Comment:
Converting it to lower case will not work as intended. Consider the sql query with a where clause with upper case criteria:

SELECT Foo.id, FOO.NAME from foo where name = 'Bar'

Reply:
Thanks for the hint